### PR TITLE
Use sensible name for URLs

### PR DESCRIPTION
**Pull request by teneightfive**
This starts to use a sensible and predictable URL structure for the feature apps.

### Auth

`/login` => `/sign-in`
`/login/signout` => `/sign-out`

### Account

`/myaccount` => `/profile`

### Support

`/support/bug` => `/support/feedback`

### Contacts

`/contact/add` => `/contacts/create`
`/contact/:id/details` => `/contacts/:id`
`/contact/:id/edit` => `/contacts/:id/edit`
`/contact/interactions/:id` => `/contacts/:id/interactions`
`/contact/:id/archive` => `/contacts/:id/archive`
`/contact/:id/unarchive` => `/contacts/:id/unarchive`

### Investment projects

`/investment/start` => `/investment-projects/create/1`
`/investment/create` => `/investment-projects/create/2`
`/investment/:id/interaction` => `/investment-projects/:id/interactions`
`/investment/:id/interaction/create` => `/investment-projects/:id/interactions/create`
`/investment/:id/interaction/:interactionId/edit` => `/investment-projects/:id/interactions/:interactionId/edit`
`/investment/:id/*` => `/investment-projects/:id/*`

### Company interactions

`/interaction/add-step-1` => `/interactions/create/1`
`/interaction/add` => `/interactions/create/2`
`/interaction/:id/details` => `/interactions/:id`
`/interaction/:id/edit` => `/interactions/:id/edit`

## Issues

Addresses #167.

### CODE REVIEW

File: src/apps/companies/controllers/interactions.js [hunk 1]
```diff
@@ -17,11 +17,11 @@ async function getInteractions (req, res, next) {
 
     // Only allow a link to add an interaction if the company has contacts
     if (company.id && company.contacts && company.contacts.length > 0) {
-      res.locals.addInteractionUrl = `/interaction/add-step-1/?company=${company.id}`
+      res.locals.addInteractionUrl = `/interactions/create/1?company=${company.id}`
```
Comment (tyom) [inline src/apps/companies/controllers/interactions.js]: I wonder if we should just always start with `create` and if there's more than one step add them in the following URLs. Or even do it on the same URL. The required data is captured in URL parameter.

Comment (teneightfive) [inline src/apps/companies/controllers/interactions.js]: Yeah, I think there's some more work that can be done specifically around creation journeys. Maybe the suggestion we just talked about as 1 URL endpoint and using params to decide which step of the journey a user is on when there are multiple steps.

Comment (teneightfive) [inline src/apps/companies/controllers/interactions.js]: Created an issue to track it #292.

File: src/apps/service-deliveries/router.js [hunk 2]
```diff
@@ -9,10 +9,10 @@ const {
 
 router.get('/servicedelivery/:serviceDeliveryId/*', getCommon)
 
-router.get('/servicedelivery/:serviceDeliveryId/details', getServiceDeliveryDetails)
+router.get('/servicedelivery/:serviceDeliveryId', getServiceDeliveryDetails)
 
 router
-  .route(['/servicedelivery/:serviceDeliveryId/edit', '/servicedelivery/edit/'])
+  .route('/servicedelivery/:serviceDeliveryId/edit')
```
Comment (tyom) [inline src/apps/service-deliveries/router.js]: `service-delivery`?

File: src/apps/companies/controllers/interactions.js (hunk 1 above)
Comment (feedmypixel) [inline src/apps/companies/controllers/interactions.js]: To me `/1` or `/2` have always felt badly named. Maybe not for this PR but can we give them more semantic names?

File: src/apps/investment-projects/controllers/create-2.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+function render (res) {
+  return res.render('investment-projects/views/create-2', {
+    title: 'Add investment project',
```
Comment (feedmypixel) [inline src/apps/investment-projects/controllers/create-2.js]: same here what does `create-2` actually mean?

Comment (teneightfive) [inline src/apps/investment-projects/controllers/create-2.js]: Agreed. The numbers in steps isn't very informative but I think we should pick that up as a separate piece of work. I created an issue (#292) to track this but think we can pick it up later.

File: src/apps/companies/controllers/interactions.js [hunk 4]
```diff
@@ -17,11 +17,11 @@ async function getInteractions (req, res, next) {
 
     // Only allow a link to add an interaction if the company has contacts
     if (company.id && company.contacts && company.contacts.length > 0) {
```
Comment (tyom) [inline src/apps/companies/controllers/interactions.js]: I know it's not your code but might change this as well.

`company.contacts` is already truthy or falsey.

File: src/apps/companies/controllers/interactions.js [hunk 5]
```diff
@@ -17,11 +17,11 @@ async function getInteractions (req, res, next) {
...
-      res.locals.addInteractionUrl = `/interaction/add-step-1/?company=${company.id}`
+      res.locals.addInteractionUrl = `/interactions/create/1?company=${company.id}`
     }
 
     if (!company.contacts || company.contacts.length === 0) {
```
Comment (tyom) [inline src/apps/companies/controllers/interactions.js]: Same here. `get(company, 'contacts', []).length`. You can cache it above somewhere.

**Pull request merged**